File: packages/serverless/lib/plugins/aws/bedrock-agentcore/examples/javascript/langgraph-basic-dockerfile/test-invoke.js
#!/usr/bin/env node
/**
 * Test script to invoke the LangGraph agent deployed to AgentCore Runtime.
 *
 * Usage:
 *   RUNTIME_ARN=arn:aws:bedrock-agentcore:... node test-invoke.js
 *
 * Or set RUNTIME_ARN environment variable before running.
 *
 * Requires: @aws-sdk/client-bedrock-agentcore
 *   npm install @aws-sdk/client-bedrock-agentcore
 */

import {
  BedrockAgentCoreClient,
  InvokeAgentRuntimeCommand,
} from '@aws-sdk/client-bedrock-agentcore'
import { randomUUID } from 'node:crypto'

const RUNTIME_ARN = process.env.RUNTIME_ARN
const REGION = process.env.AWS_REGION || 'us-east-1'
const SESSION_ID = randomUUID()

if (!RUNTIME_ARN) {
  console.error('Error: RUNTIME_ARN environment variable is required.')
  console.error('Usage: RUNTIME_ARN=<your-runtime-arn> node test-invoke.js')
  console.error('\nGet your runtime ARN from: serverless info')
  process.exit(1)
}

const client = new BedrockAgentCoreClient({ region: REGION })

async function invokeAgent(inputText) {
  console.log(`Invoking agent with input: '${inputText}'`)
  console.log(`Runtime ARN: ${RUNTIME_ARN}`)
  console.log(`Session ID: ${SESSION_ID}\n`)

  const command = new InvokeAgentRuntimeCommand({
    agentRuntimeArn: RUNTIME_ARN,
    runtimeSessionId: SESSION_ID,
    payload: Buffer.from(JSON.stringify({ prompt: inputText })),
    contentType: 'application/json',
    accept: 'application/json, text/event-stream',
  })

  const response = await client.send(command)

  console.log('Response received:')
  console.log('-'.repeat(50))

  const contentType = response.contentType || ''
  const body = response.response

  if (contentType.includes('text/event-stream')) {
    const chunks = []
    for await (const chunk of body) {
      const text =
        typeof chunk === 'string' ? chunk : new TextDecoder().decode(chunk)
      process.stdout.write(text)
      chunks.push(text)
    }
    console.log('\n')
  } else if (typeof body === 'string') {
    console.log(body)
  } else if (body && typeof body[Symbol.asyncIterator] === 'function') {
    const chunks = []
    for await (const chunk of body) {
      chunks.push(
        typeof chunk === 'string' ? chunk : new TextDecoder().decode(chunk),
      )
    }
    const text = chunks.join('')
    try {
      console.log(JSON.stringify(JSON.parse(text), null, 2))
    } catch {
      console.log(text)
    }
  } else if (body && typeof body.read === 'function') {
    const data = await body.read()
    const text = new TextDecoder().decode(data)
    try {
      console.log(JSON.stringify(JSON.parse(text), null, 2))
    } catch {
      console.log(text)
    }
  } else {
    console.log(response)
  }

  console.log('-'.repeat(50))
  console.log('Invocation completed!\n')
}

async function main() {
  console.log('LangGraph Agent Test Suite')
  console.log('='.repeat(50) + '\n')

  // Test 1: Simple calculation
  console.log('Test 1: Calculator\n')
  await invokeAgent('What is 25 multiplied by 4?')

  console.log('='.repeat(50))

  // Test 2: Time query
  console.log('\nTest 2: Time query\n')
  await invokeAgent('What is the current time?')

  console.log('='.repeat(50))
  console.log('All tests completed!')
}

main().catch((err) => {
  console.error('Test failed:', err)
  process.exit(1)
})

File: packages/serverless/lib/plugins/aws/bedrock-agentcore/examples/python/strands-browser/test-invoke.py
#!/usr/bin/env python3
"""
Test script for Strands Browser Agent

Usage:
    # Set runtime ARN from deployment output
    export RUNTIME_ARN="arn:aws:bedrock-agentcore:us-east-1:123456789012:runtime/..."

    # Run tests
    python test-invoke.py
"""

import boto3
import codecs
import json
import os
import uuid

# Get runtime ARN from environment or use default
RUNTIME_ARN = os.environ.get("RUNTIME_ARN")
if not RUNTIME_ARN:
    print("Error: RUNTIME_ARN environment variable not set")
    print("Set it to the runtime ARN from 'serverless deploy' output")
    exit(1)

# Create Bedrock AgentCore client with increased timeout for browser operations
from botocore.config import Config

config = Config(
    read_timeout=600,  # 10 minutes for browser operations
    connect_timeout=30,
    retries={'max_attempts': 1}
)
client = boto3.client("bedrock-agentcore", config=config)


def invoke_agent(prompt: str) -> dict:
    """Invoke the browser agent with a prompt."""
    session_id = str(uuid.uuid4())

    print(f"\n{'='*60}")
    print("🌐 Invoking Browser Agent")
    print(f"📍 Runtime ARN: {RUNTIME_ARN}")
    print(f"🔑 Session ID: {session_id}")
    print(f"💬 Prompt: {prompt}")
    print(f"{'='*60}\n")

    response = client.invoke_agent_runtime(
        agentRuntimeArn=RUNTIME_ARN,
        runtimeSessionId=session_id,
        payload=json.dumps({"prompt": prompt}).encode("utf-8")
    )

    # Read the streaming response (SSE) incrementally
    stream = response.get("response")
    if stream:
        final_message = None
        text_chunks = []
        decoder = codecs.getincrementaldecoder("utf-8")("replace")

        print("📡 Streaming response (this may take a while for browser operations)...")

        # Read stream in chunks
        for chunk in stream.iter_chunks():
            if chunk:
                chunk_str = decoder.decode(chunk, final=False)

                # Process lines
                for line in chunk_str.split("\n"):
                    line = line.strip()

                    if line.startswith("data:"):
                        data_str = line[5:].strip()
                        if data_str:
                            try:
                                event_data = json.loads(data_str)

                                # Extract text deltas for streaming display
                                if isinstance(event_data, dict):
                                    # Check for content delta (streaming text)
                                    if "event" in event_data:
                                        event = event_data["event"]
                                        if "contentBlockDelta" in event:
                                            delta = event["contentBlockDelta"].get("delta", {})
                                            if "text" in delta:
                                                text = delta["text"]
                                                text_chunks.append(text)
                                                print(text, end="", flush=True)

                                    # Check for final message
                                    if "message" in event_data and "content" in event_data.get("message", {}):
                                        final_message = event_data["message"]["content"]
                            except json.JSONDecodeError:
                                pass

        print("\n✅ Stream complete")

        if final_message:
            # Extract text from content array
            if isinstance(final_message, list):
                text_parts = [item.get("text", "") for item in final_message if isinstance(item, dict)]
                return {"result": "\n".join(text_parts)}
            return {"result": final_message}

        # Fall back to accumulated text chunks
        if text_chunks:
            return {"result": "".join(text_chunks)}

        return {"error": "No response content"}

    return {"error": "No response stream"}


def test_web_search():
    """Test basic web search capability."""
    print("\n🧪 Test 1: Web Search")
    result = invoke_agent(
        "Search for the latest news about AWS and summarize the top 3 headlines"
    )
    print(f"✅ Response:\n{'-'*50}")
    print(result.get("result", result))
    print(f"{'-'*50}")


def test_financial_analysis():
    """Test financial website analysis."""
    print("\n🧪 Test 2: Financial Analysis")
    result = invoke_agent(
        "Analyze the Tesla stock page at https://www.marketwatch.com/investing/stock/tsla "
        "and provide key financial metrics including current price, P/E ratio, and market cap"
    )
    print(f"✅ Response:\n{'-'*50}")
    print(result.get("result", result))
    print(f"{'-'*50}")


def test_data_extraction():
    """Test structured data extraction from a webpage."""
    print("\n🧪 Test 3: Data Extraction")
    result = invoke_agent(
        "Visit https://aws.amazon.com/about-aws/whats-new/ and list the 5 most recent announcements"
    )
    print(f"✅ Response:\n{'-'*50}")
    print(result.get("result", result))
    print(f"{'-'*50}")


if __name__ == "__main__":
    print("🚀 Strands Browser Agent Test Suite")
    print("="*60)

    # Run tests
    test_web_search()
    test_financial_analysis()
    test_data_extraction()

    print("\n" + "="*60)
    print("✅ All tests completed!")
    print("="*60)
